plotting_refactor/Dataset.py

```diff
@@ -0,0 +1,63 @@
+import time
+
+
+class Dataset:
+    def __init__(self, fitpage_index, x_data, y_data, y_fit, data_2d, plotpage_index=0):
+        self.fitpage_index = fitpage_index
+        self.x_data = x_data
+        self.y_data = y_data
+        self.y_fit = y_fit
+        self.plotpage_index = plotpage_index
+        self.data_2d = data_2d
+        self.data_id = self.generate_id(self.fitpage_index)
+
+    def generate_id(self, fitpage_index):
```

Comment:
probably need to think about unique IDs going forwards, but this a needs a proper decision between us